src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -2000,6 +2004,142 @@ public void testStats() throws DockerException, InterruptedException {
+      sut.inspectNetwork(network.id());
+      fail();
+    } catch (NetworkNotFoundException e) {
+    } catch (Exception e) {
+      fail();
+    }
+  }
+
+  @Test
+  public void testNetworksConnectContainerShouldFailIfContainerNotRunning() throws Exception {
+    assumeTrue("Docker API should be at least v1.21 to support Container Creation with " +
+                   "HostConfig, got " + sut.version().apiVersion(),
+               compareVersion(sut.version().apiVersion(), "1.21") >= 0);
+
+    assumeFalse(CIRCLECI);
+    final String networkName = randomName();
+    final String containerName = randomName();
+    final NetworkCreation networkCreation =
+        sut.createNetwork(NetworkConfig.builder().name(networkName).build());
+    assertThat(networkCreation.id(), is(notNullValue()));
+    final ContainerConfig containerConfig =
+        ContainerConfig.builder().image(BUSYBOX_LATEST).cmd("sh", "-c", "echo hello").build();
+    final ContainerCreation containerCreation = sut.createContainer(containerConfig, containerName);
+    assertThat(containerCreation.id(), is(notNullValue()));
+    try {
+      sut.connectToNetwork(containerCreation.id(), networkCreation.id());
+      fail("Expected to throw DockerException");
+    } catch (DockerException e) {
+      assertThat(e.getMessage(), containsString("is not running"));
+    } finally {
+      sut.removeContainer(containerCreation.id());
+      sut.removeNetwork(networkCreation.id());
+    }
+  }
+
+
+  @Test
+  public void testNetworksConnectContainer() throws Exception {
+    assumeTrue("Docker API should be at least v1.21 to support Container Creation with " +
+                   "HostConfig, got " + sut.version().apiVersion(),
+               compareVersion(sut.version().apiVersion(), "1.21") >= 0);
+
+    assumeFalse(CIRCLECI);
+    final String networkName = randomName();
+    final String containerName = randomName();
+    final NetworkCreation networkCreation =
+        sut.createNetwork(NetworkConfig.builder().name(networkName).build());
+    assertThat(networkCreation.id(), is(notNullValue()));
+    final ContainerConfig containerConfig =
+        ContainerConfig.builder().image(BUSYBOX_LATEST).cmd("sh", "-c", "while :; do sleep 1; done")
+            .build();
+    final ContainerCreation containerCreation = sut.createContainer(containerConfig, containerName);
+    assertThat(containerCreation.id(), is(notNullValue()));
+    sut.startContainer(containerCreation.id());
+    sut.connectToNetwork(containerCreation.id(), networkCreation.id());
+    Network network = sut.inspectNetwork(networkCreation.id());
+
+    assertThat(network.containers().size(), equalTo(1));
+    assertThat(network.containers().get(containerCreation.id()), notNullValue());
+    sut.stopContainer(containerCreation.id(), 1);
+    sut.removeContainer(containerCreation.id());
+    sut.removeNetwork(networkCreation.id());
+
+  }
+
+
+  @Test
+  public void testNetworksDisconnectContainer() throws Exception {
```

Comment:
Would it make sense to merge this test with the one above? The only additional thing tested is that the container can disconnect from the network.